plugins/catalog/src/alpha/DefaultEntityContentLayout.tsx

```diff
@@ -0,0 +1,133 @@
+import { EntitySwitch } from '../components/EntitySwitch';
+import {
+  EntityOrphanWarning,
+  isOrphan,
+} from '../components/EntityOrphanWarning';
+import {
+  EntityRelationWarning,
+  hasRelationWarnings,
+} from '../components/EntityRelationWarning';
+import {
+  EntityProcessingErrorsPanel,
+  hasCatalogProcessingErrors,
+} from '../components/EntityProcessingErrorsPanel';
+
+const useStyles = makeStyles(theme => ({
+  [theme.breakpoints.up('sm')]: {
+    infoArea: {
+      order: 1,
+    },
+    card: {
+      alignSelf: 'stretch',
+      '& > *': {
+        height: '100%',
+        minHeight: 400,
+      },
+    },
+  },
+}));
+
+const entityWarningContent = (
+  <>
+    <EntitySwitch>
+      <EntitySwitch.Case if={isOrphan}>
+        <Grid item xs={12}>
+          <EntityOrphanWarning />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+
+    <EntitySwitch>
+      <EntitySwitch.Case if={hasRelationWarnings}>
+        <Grid item xs={12}>
+          <EntityRelationWarning />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+
+    <EntitySwitch>
+      <EntitySwitch.Case if={hasCatalogProcessingErrors}>
+        <Grid item xs={12}>
+          <EntityProcessingErrorsPanel />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+  </>
+);
+
+export function DefaultEntityContentLayout(props: EntityContentLayoutProps) {
+  const { cards } = props;
+  const classes = useStyles();
+
+  const infoCards = cards.filter(card => card.type === 'info');
+  const peekCards = cards.filter(card => card.type === 'peek');
+  const fullCards = cards.filter(card => !card.type || card.type === 'full');
+  return (
+    <Grid container spacing={3}>
+      {entityWarningContent}
+      {infoCards.length > 0 ? (
+        <Grid
+          className={classes.infoArea}
+          xs={12}
+          md={4}
+          style={{
+            position: 'sticky',
+            top: -16,
+            alignSelf: 'flex-start',
+          }}
+          item
+        >
+          <Grid container spacing={3}>
+            {infoCards.map((card, index) => (
+              <Grid key={index} xs={12} item>
+                {card.element}
+              </Grid>
+            ))}
+          </Grid>
+        </Grid>
+      ) : null}
+      {peekCards.length > 0 ? (
+        <Grid xs={12} md={infoCards.length ? 8 : undefined} item>
+          <Grid container spacing={3}>
+            {peekCards.map((card, index) => (
+              <Grid key={index} className={classes.card} xs={12} md={6} item>
+                {card.element}
+              </Grid>
+            ))}
+          </Grid>
+        </Grid>
+      ) : null}
+      {fullCards.length > 0 ? (
+        <Grid xs={12} md={infoCards.length ? 8 : undefined} item>
+          <Grid container spacing={3}>
+            {fullCards.map((card, index) => (
+              <Grid key={index} className={classes.card} xs={12} md={6} item>
```

Comment:
Kinda imagining that these will always be 12, regardless of size. Maybe if you expand to widescreen we'd have some side by side, but that'd be `xxxl={6}`

Reply:
I used `md={6}` for visual backwards compatibility. Wouldn't it be a breaking change if I changed this for everyone using the default layout?

Reply:
ah, arguably not or even the other way around I'd say. Currently we're kinda switching to 3 columns instead of 2 in total, but if we keep these cards at 12 it'd be more like 1.2 + 0.8 columns :p

Reply:
In my mind, we only switch to 3 columns if people add types to the cards, but if they don't, the layout will remain 2 columns. However, I updated it to the size you suggested.